src/Microsoft.SqlTools.ServiceLayer/Utility/CommandOptions.cs

```diff
@@ -26,18 +28,31 @@ public CommandOptions(string[] args)
                     string arg = args[i];
                     if (arg.StartsWith("--") || arg.StartsWith("-"))
                     {
+                        // Extracting arguments and properties
                         arg = arg.Substring(1).ToLowerInvariant();
-                        switch (arg)
+                        string argName = arg;
+                        string argProperty = "";
+                        int splitIndex = arg.IndexOf(' ');
```

Comment:
So this code splits arguments by " "? We would only get an argument like this if we get `sqltoolsservice.exe "some argument"`. What sort of scenario do we expect this to happen?